src/main/java/com/redhat/devtools/lsp4ij/features/diagnostics/LSPDiagnosticAnnotator.java

```diff
@@ -66,28 +91,86 @@ public void doApply(@NotNull PsiFile file, Boolean applyAnnotator, @NotNull Anno
             return;
         }
         URI fileUri = LSPIJUtils.toUri(file);
-        Document document = LSPIJUtils.getDocument(file.getVirtualFile());
+        VirtualFile virtualFile = file.getVirtualFile();
+        if (virtualFile == null) {
+            return;
+        }
+        Document document = LSPIJUtils.getDocument(virtualFile);
+        if (document == null) {
+            return;
+        }
 
         // Loop for language server which report diagnostics for the given file
-        var servers = LanguageServiceAccessor.getInstance(file.getProject())
-                .getStartedServers();
+        var servers = LanguageServiceAccessor.getInstance(file.getProject()).getStartedServers();
         for (var ls : servers) {
-            LSPVirtualFileData data = ls.getLSPVirtualFileData(fileUri);
-            if (data != null) {
-                // The file is mapped with the current language server
-                var ds = data.getDiagnosticsForServer();
-                // Loop for LSP diagnostics to transform it to Intellij annotation.
-                for (Diagnostic diagnostic : ds.getDiagnostics()) {
-                    ProgressManager.checkCanceled();
-                    createAnnotation(diagnostic, document, file, ds, holder);
+            // TODO: Is there not a way to check whether the server is valid for this file here?

```

Comment:
I noticed that `LSPDiagnosticFeature#isSupported()` always returns `true`, so this block is being executed for language servers that don't support the file. Is there some way to check whether the language server supports the file, e.g., based on its mapping configuration?